client/my-sites/post-share/index.jsx

```diff
@@ -51,11 +58,28 @@ class PostShare extends Component {
 		requestConnections: PropTypes.func
 	};
 
+	static FOOTER_SECTION_SCHEDULED = 'footer-section-scheduled';
+	static FOOTER_SECTION_PUBLISHED = 'footer-section-published';
+
 	state = {
 		message: PostMetadata.publicizeMessage( this.props.post ) || this.props.post.title,
 		skipped: PostMetadata.publicizeSkipped( this.props.post ) || [],
+		footerSection: PostShare.FOOTER_SECTION_SCHEDULED
+
 	};
 
+	constructor() {
+		super( ...arguments );
+
+		this.setFooterSection = this.setFooterSection.bind( this );
+	}
+
+	setFooterSection( section ) {
```

Comment:
Let's bind the function on the fly. It allow us:
* do not add the constructor to the class
* do no use `partial` lodash function

```es6
setFooterSection = section => {
  // ...
}; // do not forget the semicolon here
```